### PR TITLE
feat: add `eth_getBlockByNumber` support

### PR DESCRIPTION
**Pull request by agostbiro**
Adds `eth_getBlockByNumber` support to the EDR provider, matching the Hardhat implementation. 

Hardhat integration tests are passing up to the point where they need `eth_sendTransaction`.

### CODE REVIEW

**Review by Wodann — CHANGES_REQUESTED** [review]
Mostly LGTM. A couple of comments for clarification

File: crates/edr_evm/src/block/remote.rs [hunk 1]
```diff
@@ -97,15 +103,21 @@ impl RemoteBlock {
 
         let hash = block.hash.ok_or(CreationError::MissingHash)?;
 
+        let size = block
```
[the remark this answers is not in the document]
Comment (agostbiro) [reply crates/edr_evm/src/block/remote.rs]: Doesn't mandate it as far as I can tell. I changed the size type from `U256` to `u64` on the remote type: [c40e020](https://github.com/NomicFoundation/hardhat/pull/4627/commits/c40e0206e4424773d7ef59a8931f093e84cc95ad)

File: crates/edr_evm/src/block/remote.rs [hunk 2]
```diff
@@ -45,10 +48,13 @@ pub struct RemoteBlock {
...
     /// The hashes of the block's ommers
-    _ommers: Vec<B256>,
-    _withdrawals: Option<Vec<Withdrawal>>,
+    ommer_hashes: Vec<B256>,
+    /// The staking withdrawals
+    withdrawals: Option<Vec<Withdrawal>>,
     /// The block's hash
     hash: B256,
+    /// Integer size of this block in bytes
+    size: u64,
```
[the remark this answers is not in the document]
Comment (agostbiro) [reply crates/edr_evm/src/block/remote.rs:57]: Done in [c40e020](https://github.com/NomicFoundation/hardhat/pull/4627/commits/c40e0206e4424773d7ef59a8931f093e84cc95ad)

File: crates/edr_evm/src/block.rs [hunk 3]
```diff
@@ -25,6 +28,12 @@ pub trait Block: Debug {
     /// Returns the block's header.
     fn header(&self) -> &block::Header;
 
+    /// Uncle block hashes.
+    fn ommer_hashes(&self) -> &[B256];
+
+    /// The size of this block in bytes.
+    fn size(&self) -> u64;
```
[the remark this answers is not in the document]
Comment (agostbiro) [reply crates/edr_evm/src/block.rs]: Good idea, done in [c40e020](https://github.com/NomicFoundation/hardhat/pull/4627/commits/c40e0206e4424773d7ef59a8931f093e84cc95ad)

**Review by Wodann — APPROVED** [review]

**Pull request merged**